items/sign_host_keys.py

```diff
@@ -0,0 +1,145 @@
+import os.path
+from datetime import timedelta, datetime
+from pathlib import Path
+from tempfile import mkdtemp
+
+import bundlewrap.exceptions
+from bundlewrap.items import Item
+from bundlewrap.utils.remote import PathInfo
+
+try:
+    from sshkey_tools.cert import SSHCertificate
+    from sshkey_tools.keys import PrivateKey, PublicKey
+except ImportError:
+    raise bundlewrap.exceptions.BundleError("Please install package sshkey-tools>=0.9 first.")
+
+
+# See https://stackoverflow.com/a/49782093
+def remove_dir_recursive(path):
+    directory = Path(path)
+    for item in directory.iterdir():
+        if item.is_dir():
+            os.rmdir(item)
+        else:
+            item.unlink()
+    directory.rmdir()
+
+
+class SignHostKeys(Item):
+    """
+    Sign SSH Host Keys
+    """
+    BUNDLE_ATTRIBUTE_NAME = "sign_host_keys"
+    NEEDS_STATIC = [
+        "pkg_apt:",
+        "pkg_pacman:",
+        "pkg_yum:",
+        "pkg_zypper:",
+    ]
+    ITEM_ATTRIBUTES = {
+        'path': None,
+        'ca_password': None,
+        'ca_path': None,
+        'days_valid': 3650,
+    }
+    ITEM_TYPE_NAME = "sign_host_key"
+    REQUIRED_ATTRIBUTES = [
+        'path',
+        'ca_password',
+        'ca_path',
+    ]
+
+    def get_cert_path(self):
+        return self.attributes.get('path') + '.pub.crt'
+
+    @classmethod
+    def block_concurrent(cls, node_os, node_os_version):
+        """
+        Return a list of item types that cannot be applied in parallel
+        with this item type.
+        """
+        return []
+
+    def __repr__(self):
+        return "<Sign Host Key path:{} ca_path:{}>".format(self.attributes['path'],
+                                                                 self.attributes['ca_path'])
+
+    def cdict(self):
+        return {
+            f'{self.get_cert_path()} exist': True
+        }
+
+    def sdict(self):
+        path_info = PathInfo(self.node, self.get_cert_path())
+        return {
+            f'{self.get_cert_path()} exist': path_info.exists
```

Comment:
does it makes sense to verify the signature? to make sure, this can be updated? but will most probably make the system more complex. Then instead in the readme we should state, in case of rekey, please delete the files on host via bw run 'rm ...'

Reply:
Don't think so, should be enough to just recreate the key at each run. You could use `unless` in your `items.py` to prevent double execution.